### ISSUE TITLE
Make geohash coordinates lay inside polygons

### ISSUE THREAD

**Issue by CEKrause** [GeoscienceAustralia/dea-waterbodies#12]
At the moment, geohashes are applied to the centroids of polygons. If a polygon is 'U'-shaped, then the centroid for the polygon will occur outside of the polygon itself, and the geohash will not make it clear which polygon it refers to. 

Change the way the geohash is applied and apply it to the centroid of the polygon AND inside the polygon.

**Comment by MatthewJA** [comment]
What kind of method were you thinking for this? Project centroid to the nearest boundary vertex?

**Comment by CEKrause** [comment]
@vnewey had some ideas for how we could implement this. I am not sure how easy it would be to do since we are using the geohash python library to attribute geohashes. It would be worth seeing if the library has a switch that already does this? Otherwise we may want to look into doing a PR into their repo to implement?

**Comment by CEKrause** [comment]
This is the repo for the function we use https://github.com/hkwi/python-geohash

**Comment by CEKrause** [comment]
hmmm. This repo looks unmaintained. Maybe we should explore if there's a better option?

**Comment by MatthewJA** [comment]
Agreed! Maybe [geohash2](https://github.com/dbarthe/geohash/)? I think this one is available on Sandbox, too.

**Comment by MatthewJA** [comment]
Though I don't think there's anything wrong with either module... all geohash modules seem to not have been updated recently
